### PR TITLE
ci: release apple arm binary

### PR DESCRIPTION
**Pull request by MarcoIeni**
Requested [here](https://github.com/dandavison/delta/issues/484#issuecomment-1312790539)

There are no VMs for GitHub actions that support Mac ARM. See https://github.com/actions/runner-images/issues/2187.
So we cannot run tests for MAC ARM. However, we can run `cargo build`, which is what's required to build the binary.

- [x] to be merged after #1238

### CODE REVIEW

**Comment by MarcoIeni** [comment]
```
Caused by:
  Bad CPU type in executable (os error 86)
```

I need to fix this.

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -12,19 +12,37 @@ jobs:
...
+          - os: ubuntu-latest
+            target: aarch64-unknown-linux-gnu
+            use-cross: true
 
     steps:
       - name: Installing Rust toolchain
         uses: actions-rs/toolchain@v1
         with:
           toolchain: stable
+          target: ${{ matrix.job.target }}
```
Comment (MarcoIeni) [inline .github/workflows/cd.yml:45]: This installs the target if missing. Useful for `aarch64-apple-darwin` on mac

**Comment by MarcoIeni** [comment]
This is ready now. 👍

**Comment by dandavison** [comment]
Hi @MarcoIeni, thanks for this work!

So basically, we add `aarch64-apple-darwin` to cd but not to ci. Is that an accurate summary?

I've merged your reformatting PR so I'll wait for you to rebase this one.

**Comment by MarcoIeni** [comment]
> So basically, we add aarch64-apple-darwin to cd but not to ci. Is that an accurate summary?

Yes, exactly

**Comment by dandavison** [comment]
Thank you very much @MarcoIeni!

**Pull request merged**